From 4a0a965547b52afc953d9bd11ab059f543551459 Mon Sep 17 00:00:00 2001
From: Fabian Greffrath <fabian@greffrath.com>
Date: Tue, 16 Sep 2014 19:02:37 +0200
Subject: [PATCH] fix rabbit disappearing in upper left corner

aka "undesired springs added above the screen"

If the bunny jumps out of the screen, its
y-coordinate becomes negative. Thus, the array that is checked for the
position of the objects on the map overflows and returns some bogus,
e.g. that there are springs whereas there aren't. This patch
checks for the y-coordinate and caps it at zero.

c.f. https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=370650
---
 main.c | 2 ++
 1 file changed, 2 insertions(+)

--- a/main.c
+++ b/main.c
@@ -2096,6 +2096,8 @@ void steer_players(void)
 
 				s1 = (player[c1].x >> 16);
 				s2 = (player[c1].y >> 16);
+				if (s2 < 0)
+					s2 = 0;
 				if (ban_map[(s2 + 15) >> 4][(s1 + 8) >> 4] == BAN_SPRING || ((ban_map[(s2 + 15) >> 4][s1 >> 4] == BAN_SPRING && ban_map[(s2 + 15) >> 4][(s1 + 15) >> 4] != BAN_SOLID) || (ban_map[(s2 + 15) >> 4][s1 >> 4] != BAN_SOLID && ban_map[(s2 + 15) >> 4][(s1 + 15) >> 4] == BAN_SPRING))) {
 					player[c1].y = ((player[c1].y >> 16) & 0xfff0) << 16;
 					player[c1].y_add = -400000L;
